R/indexing.R

```diff
@@ -322,3 +357,303 @@ BasicIndexer <- R6::R6Class("BasicIndexer",
+                                    if(!is_na(p$dim_out_sel)) {
+                                      # out_sel <- append(out_sel, p$dim_out_sel)
+                                      out_sel <- append(out_sel, list(p$dim_out_sel))
+                                    }
+                                  }
+                                  
+                                  result <- append(result, ChunkProjection$new(
+                                    chunk_coords,
+                                    chunk_sel,
+                                    out_sel
+                                  ))
+                                }
+                                
+                                return(result)
+                              }
+                            )
+)
+
+# Reference: https://github.com/zarr-developers/zarr-python/blob/5dd4a0e6cdc04c6413e14f57f61d389972ea937c/zarr/indexing.py#L424
+
+#' The Order class.
+#' @title Order Class
+#' @docType class
+#' @description
+#'  TODO
+#' @rdname Order
+#' @keywords internal
+Order <- R6::R6Class("Order",
+                     public = list(
+                       #' @field UNKNOWN UNKNOWN
+                       #' @keywords internal
+                       UNKNOWN = 0,
+                       #' @field INCREASING INCREASING
+                       #' @keywords internal
+                       INCREASING = 1,
+                       #' @field DECREASING DECREASING
+                       #' @keywords internal
+                       DECREASING = 2,
+                       #' @field UNORDERED UNORDERED
+                       #' @keywords internal
+                       UNORDERED = 3,
+                       #' @description
+                       #' checking order of numbers
+                       #' @param a vector of numbers
+                       check = function(a){
+                         diff_a <- diff(a)
+                         diff_positive <- diff_a >= 0
+                         n_diff_positive <- sum(diff_positive)
+                         all_increasing <- n_diff_positive == length(diff_positive)
+                         any_increasing <- n_diff_positive > 0
+                         if(all_increasing){
+                           return(Order$public_fields$INCREASING)
+                         } else if(any_increasing) {
+                           return(Order$public_fields$UNORDERED)
+                         } else{
+                           return(Order$public_fields$DECREASING)
+                         }
+                       })
+                     )
+
+# Reference: https://github.com/zarr-developers/zarr-python/blob/5dd4a0e6cdc04c6413e14f57f61d389972ea937c/zarr/indexing.py#L457
+
+#' The Zarr IntArrayDimIndexer class.
+#' @title IntArrayDimIndexer Class
+#' @docType class
+#' @description
+#'  TODO
+#' @rdname IntArrayDimIndexer
+#' @keywords internal
+IntArrayDimIndexer <- R6::R6Class("IntArrayDimIndexer",
```

Comment:
Similarly, would you be able to add a test for this `IntArrayDimIndexer` class?

Reply:
Some more tests for `IntArrayDimIndexer` alongside with `OrthogonalIndexers` 6f5d6c276f43d2ee63bc9721eb866f8eb86971ca